src/formatter/helpers/issue_helpers.js

```diff
@@ -51,67 +53,112 @@ function formatDataTable(dataTable) {
   return table.toString()
 }
 
-function formatDocString(docString) {
-  return '"""\n' + docString.content + '\n"""'
+function formatDocString(arg) {
+  return '"""\n' + arg.content + '\n"""'
 }
 
-function formatStepResult({ colorFns, stepResult }) {
-  const { status, step } = stepResult
+function formatStep({
+  colorFns,
+  isBeforeHook,
+  keyword,
+  keywordType,
+  pickleStep,
+  snippetBuilder,
+  testStep
+}) {
+  const { status } = testStep.result
   const colorFn = colorFns[status]
 
-  const symbol = CHARACTERS[stepResult.status]
-  const identifier = colorFn(symbol + ' ' + step.keyword + (step.name || ''))
-  let text = identifier
+  let identifier
+  if (testStep.sourceLocation) {
+    identifier = keyword + (pickleStep.text || '')
+  } else {
+    identifier = isBeforeHook ? 'Before' : 'After'
+  }
+
+  let text = colorFn(CHARACTERS[status] + ' ' + identifier)
 
-  const { stepDefinition } = stepResult
-  if (stepDefinition) {
-    const stepDefinitionLocation = formatLocation(stepDefinition)
-    const stepDefinitionLine = ' # ' + colorFns.location(stepDefinitionLocation)
-    text += stepDefinitionLine
+  const { actionLocation } = testStep
+  if (actionLocation) {
+    text += ' # ' + colorFns.location(formatLocation(actionLocation))
   }
   text += '\n'
 
-  _.each(step.arguments, arg => {
+  if (pickleStep) {
     let str
-    if (arg instanceof DataTable) {
-      str = formatDataTable(arg)
-    } else if (arg instanceof DocString) {
-      str = formatDocString(arg)
-    } else {
-      throw new Error('Unknown argument type: ' + arg)
+    const iterator = buildStepArgumentIterator({
+      dataTable: arg => (str = formatDataTable(arg)),
+      docString: arg => (str = formatDocString(arg))
+    })
+    _.each(pickleStep.arguments, iterator)
+    if (str) {
+      text += indentString(colorFn(str) + '\n', 4)
     }
-    text += indentString(colorFn(str) + '\n', 4)
+  }
+  const message = getStepMessage({
+    colorFns,
+    keywordType,
+    pickleStep,
+    snippetBuilder,
+    testStep
   })
+  if (message) {
+    text += indentString(message, 4) + '\n'
+  }
   return text
 }
 
+export function isIssue(status) {
+  return _.includes(
+    [Status.AMBIGUOUS, Status.FAILED, Status.PENDING, Status.UNDEFINED],
+    status
```

Comment:
This can perhaps be a object with keys to look up, using JS `in`? Or am I missing something important?